crates/containerd-shim-wasmtime/src/executor.rs

```diff
@@ -0,0 +1,133 @@
+use std::{fs::OpenOptions, os::fd::RawFd, path::PathBuf};
+
+use anyhow::{anyhow, Result};
+use containerd_shim_wasm::sandbox::oci;
+use libc::{dup, dup2, STDERR_FILENO, STDIN_FILENO, STDOUT_FILENO};
+use libcontainer::workload::{Executor, ExecutorError};
+use oci_spec::runtime::Spec;
+
+use wasmtime::{Engine, Linker, Module, Store};
+use wasmtime_wasi::WasiCtxBuilder;
+
+use crate::oci_wasmtime::{self, wasi_dir, wasi_file};
+
+const EXECUTOR_NAME: &str = "wasmtime";
+
+static mut STDIN_FD: Option<RawFd> = None;
+static mut STDOUT_FD: Option<RawFd> = None;
+static mut STDERR_FD: Option<RawFd> = None;
+
+pub struct WasmtimeExecutor {
+    pub stdin: Option<RawFd>,
+    pub stdout: Option<RawFd>,
+    pub stderr: Option<RawFd>,
+    pub engine: Engine,
+}
+
+impl Executor for WasmtimeExecutor {
+    fn exec(&self, spec: &Spec) -> Result<(), ExecutorError> {
+        let args = oci::get_args(spec);
+        if args.is_empty() {
+            return Err(ExecutorError::InvalidArg);
+        }
+
+        let (mut store, f) = self
+            .prepare_function(spec, args)
+            .map_err(|err| ExecutorError::Other(format!("failed to prepare function: {}", err)))?;
+
+        log::info!("calling start function");
+        match f.call(&mut store, &[], &mut []) {
+            Ok(_) => std::process::exit(0),
+            Err(_) => std::process::exit(137),
+        };
+    }
+
+    fn can_handle(&self, _spec: &Spec) -> bool {
+        true
+    }
+
+    fn name(&self) -> &'static str {
+        EXECUTOR_NAME
+    }
+}
+
+impl WasmtimeExecutor {
+    fn prepare_function(
```

Comment:
We know at a glance that this is a function.
```suggestion
    fn prepare(
```